### PR TITLE
Filter BOM bytes in the readerWithDetectedCharset

### PR DESCRIPTION
**Pull request by er1c**

### CODE REVIEW

**Comment by tpunder** [comment]
Probably need some tests for this

File: src/main/scala/fm/common/IOUtils.scala [hunk 1]
```diff
@@ -101,6 +101,13 @@ object IOUtils {
    */
   def read(input: Reader, buffer: Array[Char], offset: Int, length: Int): Int = ApacheIOUtils.read(input, buffer, offset, length)
   
+  // Read all of the characters from a reader into a string
+  def readToString(reader: Reader): String = {
+    val writer = new StringWriter()
+    IOUtils.copy(reader, writer)
+    writer.toString
+  }
+
```
Comment (tpunder) [inline src/main/scala/fm/common/IOUtils.scala]: How is this different from the ```toString(input: Reader)``` method?

Comment (er1c) [inline src/main/scala/fm/common/IOUtils.scala]: Not at all, I was looking for a method but missed that toString call

**Comment by er1c** [comment]
PR Updated, removed readToString

**Comment by tpunder** [comment]
Can you add sample files & tests for ISO-8859-1, MacRoman, and Windows-1252.  I think we see all of those encodings.

**Comment by er1c** [comment]
PR updated, I updated `reader(encoding: String)` to call `reader(charset: Charset)` and `reader(charset: Charset)` has the BOM byte filtering

**Pull request merged**